tutorials/nextjs-app-serverless-functions-sqldb/index.mdx

```diff
@@ -0,0 +1,948 @@
+
+<Message type="tip">
+If you want to avoid doing these modifications manually, you can also directly clone the final code from this tutorial from a remote repository with `git clone git@gitlab.infra.online.net:fpagny/tutorial-sdb-nextjs-terraform.git`. Note that this repository differs slightly in configuration, as it will dynamically render pages and not pre-render them in the Nextjs app build phase.
+</Message> 
+
+1. Edit the file located at `src/lib/api.ts` by replacing its content with the following code. This will switch frontend content from local data to your **Serverless SQL Database**.
+    ```typescript
+    import { Post } from "@/interfaces/post";
+    import fs from "fs";
+    import matter from "gray-matter";
+    import { join } from "path";
+    import { Client } from 'pg'
+
+
+    export type Row = {
+      slug: string;
+      title: string;
+      date: Date;
+      coverimage: string;
+      author_name: string;
+      author_picture: string;
+      excerpt: string;
+      ogimage_url: string;
+      content: string;
+      preview?: boolean;
+    };
+
+    const client = new Client({
+        ssl: {
+          rejectUnauthorized: false,
+        }
+      })
+
+    const connect = async () => {
+      await client.connect()
+    }
+
+    connect()
+
+    //This default function handles Nextjs pre-rendering with undefined values
+    export function getPostFromRow(row: Row){
+      if (row === undefined){ 
+        var postRow: Row = {
+          slug: "Title",
+          title: "Excerpt",
+          date: new Date("2024-01-16"),
+          coverimage: "",
+          author_name: "",
+          author_picture: "",
+          excerpt: "",
+          ogimage_url: "",
+          content: "",
+          preview: false
+        } 
+      } else{
+        var postRow = row
+      }
+
+      const post = {
+        title: postRow.title,
+        excerpt: postRow.excerpt,
+        coverImage: postRow.coverimage,
+        date: postRow.date.toISOString(),
+        author: {
+          name: postRow.author_name,
+          picture: postRow.author_picture
+        },
+        ogImage: {
+          url: postRow.ogimage_url
+        },
+        slug: postRow.slug,
+        content: postRow.content,
+        preview: false
+      }
+      return post;
+    }
+
+
+    export async function getPostBySlug(slug: string): Promise<Post> {
+      const data = await client.query('SELECT * FROM posts WHERE slug=$1;',[slug])
+      const post = getPostFromRow(data.rows[0])
+      return post;
+    }
+
+
+    export async function getAllPosts(): Promise<Post[]> {
+      const data = await client.query('SELECT * FROM posts;')
+      const posts = data.rows.map((row) => (getPostFromRow(row)))
+      return posts;
+    }
+    ```
+
+2. edit the file located at `src/app/page.tsx` by replacing its content with the following code. This will update the main application home page which displays the posts list.
```

Comment:
```suggestion
2. Edit the file located at `src/app/page.tsx` by replacing its content with the following code. This will update the main application home page which displays the posts list.
```